Makefile

```diff
@@ -88,26 +88,27 @@ build.all:
 
 # ==============================================================================
 # Clean targets
-.PHONY: clean clean.nones clean.all clean.nuke
+.PHONY: clean clean.nones clean.all
 
 clean.image.%: do.image.%
 	@$(call set_env_vars); rm -rf $$STAGING_DIR
 
 clean: $(foreach p, $(FLAVORS), clean.image.$(HOST_ARCH),$(p))
 
 clean.nones:
-	@docker rmi -f $(shell docker images | egrep "^<none> " | awk '{print $$3}' | uniq) || true
+	@if docker images | egrep -q "^<none> "; then \
+		docker rmi -f $(shell docker images | egrep "^<none> " | awk '{print $$3}' | uniq) || true ; \
+	fi
 
 clean.all: clean.nones
 	@rm -rf staging/
-	# Don't mess with other registries for some semblance of a safe clean.
-	@docker rmi -f \
-		$(shell docker images | egrep "^$(TAG_REGISTRY)/daemon(-base)? " | \
-		  awk '{print $$3}' | uniq) || true
-
-clean.nuke: clean.all
-	@docker rmi -f \
-		$(shell docker images | egrep "^.*/daemon(-base)? " | awk '{print $$3}' | uniq) || true
+	@# Inspect each image, and if we find 'CEPH_POINT_RELEASE' we can be pretty sure it's a
+	@# ceph-container image and safe to delete
+	@for image in $(shell docker images | grep ago | awk '{ print $$3 }'); do \
+		if $$(docker inspect "$$image" | grep -q 'CEPH_POINT_RELEASE'); then \
```

Comment:
if you want to test whether the command returns true or false you don't need a subshell for this and simply do:

`if docker inspect "$$image" | grep -q 'CEPH_POINT_RELEASE'; then`

Unless this has something to do with make...

Reply:
It might have something to do with make ... or I might've just overlooked that. I'll check.